### PR TITLE
feat: Add custom transformer support to PCAPredictor

### PR DESCRIPTION
**Pull request by melund**
This PR updates PCAPredictor to allow using custom scikit-learn transformers instead of always defaulting to StandardScaler. This makes it possible to use other scalers (like MinMaxScaler) or full pipelines (e.g., adding a PowerTransformer before PCA).

Changes:

* Added a transformer argument to PCAPredictor.
* Updated predict and parameter conversion methods to delegate scaling to the transformer.
* Updated export_pca_data to also save the transformer via pickle (includes fallback for loading old files).
* Added test_custom_transformer.py to cover the new functionality.
* Documented the feature in README.md.

### CODE REVIEW

**Comment by melund** [comment]
@JohnRasmussenAAU/@JohnRasmussenAnyBody I have tried to implement your idea with an extra transformer. It is implemented in generic way so you can use any transformation for scikitlearn. 

By default, `PCAPredictor` uses `StandardScaler` to normalize data. You can provide a custom transformer (e.g., `MinMaxScaler`, `PowerTransformer`, or a `Pipeline`) to control preprocessing. Of course it only works with transformers that works columns/element wise. 

```python
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import PowerTransformer
from sillywalk import PCAPredictor

transformer = make_pipeline(
    PowerTransformer(method="yeo-johnson", standardize=True)
)

model = PCAPredictor(df, transformer=transformer)
prediction = model.predict({"Height": 1.8})
```

**Comment by JohnRasmussenAnyBody** [comment]
Wow, that was quick. I am also playing around with this, but I like your way. Does PCAPredictor transform backwards after the fact when presented with a transformer argument?

**Comment by melund** [comment]
Yes. I am not super knowledgeable of scikitlearns transformer objects. But they seem to have a `transformer.inverse_transform()` function. Maybe some of them doesn't. I am not quite sure, if it could break in that situation.

**Comment by melund** [comment]
I had to think a bit about how to save the model, so it doens't have to be instantiated from the raw full dataset. 

Here I just added the a pickled version of the transformer to the data along with the eigenvectors etc. 

This could all be made simpler by just saving the whole object as a pickle file. But only saving the essential data, it makes it a bit independent of the the version of sillywalk.

**Comment by melund** [comment]
Scikitlearn's `pipeline.make_pipeline()` is pretty cool

**Comment by JohnRasmussenAnyBody** [comment]
If you merge this into main, will I get it automatically when I pixi run python whatever.py? If so, please tell me when I can try it out.

**Comment by melund** [comment]
You can add `preview = ["pixi-build"]` to your `pixi.toml` file to enable a preview feature, which allows you to install sillywalk directly from source by specifying the dependency like this: 
```
sillywalk = {git="https://github.com/AnyBody-Research-Group/sillywalk.git@more-preconditioning"}
```

Baiscally something like this in the 'pixi.toml' file:

```
[workspace]
channels = ["conda-forge"]
name = "example-git-dependency"
platforms = ["win-64"]
preview = ["pixi-build"] # <-- add this line
[tasks]

[dependencies]
# Then add this line to install sillywalk directly from the branch 'more-preco...":
sillywalk = {git="https://github.com/AnyBody-Research-Group/sillywalk.git@more-preconditioning"}
```

When you then do `pixi run <something>` or `pixi install` it will get the version from the branch install it.

**Comment by melund** [comment]
It will not get released to conda-forge until we add a git tag with a new version. That triggers the pipeline which pushes a release to pypi and conda-forge. 

But the trick I described before allows you to test unreleased stuff.

**Comment by melund** [comment]
@JohnRasmussenAnyBody I have rebased this PR on top of all the fixes I made in #26 #27 #28 #29 

It also now validates that the 'custom' transformer is indeed an 'element'-vise transformer like the StandardScaler and 'yeo-johnson' transformer.
